test/Microsoft.ApplicationInsights.AspNetCore.Tests/Extensions/ApplicationInsightsExtensionsTests.cs

```diff
@@ -201,15 +215,38 @@ public static void RegistersTelemetryConfigurationFactoryMethodThatReadsInstrume
                 }
             }
 
+            /// <summary>
+            /// Validates that while using services.AddApplicationInsightsTelemetry(); ikey is read from
+            /// Environment
+            /// </summary>
+            [Fact]
+            [Trait("Trait", "ConnectionString")]
+            public static void AddApplicationInsightsTelemetry_ReadsConnectionString_FromEnvironment()
+            {
+                var services = ApplicationInsightsExtensionsTests.GetServiceCollectionWithContextAccessor();
+                Environment.SetEnvironmentVariable(ConnectionStringEnvironmentVariable, TestConnectionString);
+                try
+                {
+                    services.AddApplicationInsightsTelemetry();
+                    IServiceProvider serviceProvider = services.BuildServiceProvider();
+                    var telemetryConfiguration = serviceProvider.GetTelemetryConfiguration();
+                    Assert.Equal(TestConnectionString, telemetryConfiguration.ConnectionString);
+                }
+                finally
+                {
+                    Environment.SetEnvironmentVariable(ConnectionStringEnvironmentVariable, null);
+                }
+            }
+
             /// <summary>
             /// Validates that while using services.AddApplicationInsightsTelemetry(); ikey is read from
             /// Environment
             /// </summary>
             [Fact]
             public static void AddApplicationInsightsTelemetryReadsInstrumentationKeyFromEnvironment()
             {
-                var services = ApplicationInsightsExtensionsTests.GetServiceCollectionWithContextAccessor();                
-                Environment.SetEnvironmentVariable("APPINSIGHTS_INSTRUMENTATIONKEY", TestInstrumentationKey);                
+                var services = ApplicationInsightsExtensionsTests.GetServiceCollectionWithContextAccessor();
```

Comment:
lets add one test with both ikey,endpoint in appsettings.json AND connectionstring also on appsettings.json, Validate that connectionstring Wins.